README.md

```diff
@@ -8,60 +8,122 @@
 
-### **`useLocalStore<T, S>(initializer: () => T, source?: S, annotations?): T`**
+### **`useLocalStore<T, S>(initializer: () => T, source?: S): T`** (deprecated)
+
+_This API is deprecated in 3.\*. Use `useLocalObservable` instead. They do roughly the same, but `useLocalObservable` accepts an set of annotations as second argument, rather than a `source` object. Using `source` is not recommended, see the deprecation message at `useAsObservableSource` for details_
 
 Local observable state can be introduced by using the useLocalStore hook, that runs its initializer function once to create an observable store and keeps it around for a lifetime of a component.
 
 The annotations are similar to the annotations that are passed in to MobX's [`observable`](https://mobx.js.org/observable.html#available-annotations) API, and can be used to override the automatic member inference of specific fields.
 
-### **`useAsObservableSource<T>(source: T): T`**
+### **`useAsObservableSource<T>(source: T): T`** (deprecated)
 
 The useAsObservableSource hook can be used to turn any set of values into an observable object that has a stable reference (the same object is returned every time from the hook).
 
-## Observer batching
+_This API is deprecated in 3.\* as it relies on observables to be updated during rendering which is an anti-pattern. Instead, use `useEffect` to synchronize non-observable values with values. Example:_
+
+```javascript
+// Before:
+function Measurement({ unit }) {
+    const observableProps = useAsObservableSource({ unit })
+    const state = useLocalStore(() => ({
+        length: 0,
+        get lengthWithUnit() {
+            // lengthWithUnit can only depend on observables, hence the above conversion with `useAsObservableSource`
+            return observableProps.unit === "inch"
+                ? `${this.length * 2.54} inch`
+                : `${this.length} cm`
+        }
+    }))
+
+    return <h1>{state.lengthWithUnit}</h1>
+}
+
+// After:
+function Measurement({ unit }) {
+    const state = useLocalObservable(() => ({
+        unit, // the initial unit
+        length: 0,
+        get lengthWithUnit() {
+            // lengthWithUnit can only depend on observables, hence the above conversion with `useAsObservableSource`
+            return this.unit === "inch" ? `${this.length * 2.54} inch` : `${this.length} cm`
+        }
+    }))
+
+    useEffect(() => {
+        // sync the unit from 'props' into the observable 'state'
+        state.unit = unit
+    }, [unit])
+
+    return <h1>{state.lengthWithUnit}</h1>
+}
+```
+
+Note that, at your own risk, it is also possible to not use `useEffect`, but do `state.unit = unit` instead in the rendering.
+This is closer to the old behavior, but React will warn correctly about this if this would affect the rendering of other components.
+
+## Observer batching (deprecated)
```

Comment:
Maybe add a short explanation that this is done automatically now and they don't need to worry about it?

Reply:
See next line :)